test_util/run-all

```diff
@@ -113,8 +113,16 @@ RET=$?
 mkdir -p $WORKDIR/test/vagrant-logs
 if [ $RET -ne 0 ]; then
   echo "TEST FAILED"
+  echo "Attempting to gather logs"
+  export LOG_REPLAY_LINES=${LOG_REPLAY_LINES:=1000}
   for i in boot m1 m2 m3 a1 a2; do
-    vagrant ssh $i -c 'sudo journalctl -o short' > $WORKDIR/test/vagrant-logs/$i
+    if [ "$TEAMCITY_BUILD_ENV" == "true" ]; then
+      echo "##teamcity[blockOpened name='$i']"
+      vagrant ssh $i -c "'sudo journalctl -n $LOG_REPLAY_LINES'"
+      echo "##teamcity[blockClosed name='$i']"
+    else
+      vagrant ssh $i -c 'sudo journalctl -o short' > $WORKDIR/test/vagrant-logs/$i
```

Comment:
This commit was just merging the build steps on TC into this script to avoid situations of "something merged, so immediately go fix the build system", thus I didnt want to change the status quo.

As for what we should do, I would agree on dumping the folder. I think using short should be the default (most people dont want thousands of lines of stuff dumped to their terminal) and if LOG_REPLAY_LINES  is set (as it is in TC), then use that value